okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -894,50 +894,41 @@ public Builder newBuilder() {
   }
 
   /**
-   * Returns a new {@code HttpUrl} representing {@code url} if it is a well-formed HTTP or HTTPS
-   * URL, or null if it isn't.
+   * Returns a new {@code HttpUrl} parsed from the string value of the {@code object} if it is a
+   * well-formed HTTP or HTTPS URL, or null if it isn't.
+   *
+   * @param object for check URL format
+   * @return instance of {@link HttpUrl} or {@code null} in the case of a parse error
    */
-  public static @Nullable HttpUrl parse(String url) {
+  public static @Nullable HttpUrl parse(Object object) {
```

Comment:
Why Object as the param type?

Reply:
In our case, no matter what object pars URL, URI, InternetAddress or something else. In any case, whichever object we get, the very concept of parsing a string implies an object value of in a string.
I removed unnecessary methods that did not really make sense and duplicated the code.
~~get(URL)~~
~~get(URI)~~
etc.